### DISCUSSION TITLE
Doris&Metabase in product environment

### DISCUSSION THREAD

**Discussion by shenzlx** [apache/doris#30302] [A - General / Q&A]
有人在正式的项目或产品中，使用过 DORIS + METABASE 的搭配吗？

**Comment by HWTsang** [comment]
我们就是这样用的, 目前没问题

**Reply by ihadoop** [reply]
有啥不支持的数据类型或者功能码

**Reply by HWTsang** [reply]
目前没有遇到过, 分析类场景, 速度非常快. 不支持的可能是我们当时想把1000+个字段放到json类型的数据里面, 用的时候直接解包来用, 当时会导致内存溢出, 后来还是删掉了, 改成正常表结构就没问题

**Reply by ihadoop** [reply]
超大的json确实会oom，最近在写metabase的doris driver（暂时参考mysql的driver）。所以问问如果有不支持的字段类型，或者遇到问题可以通过driver来适配。

**Reply by HWTsang** [reply]
我们metabase直接按连接mysql的方式连接的doris, 然后没有写sql配置, 是用的选项配置的

**Reply by ihadoop** [reply]
问下，你用的doris版本是多少？
我用metabase的mysql驱动测试的时候，创建数据源会提示
<img width="1312" alt="image" src="https://github.com/apache/doris/assets/5036560/6ed5f66d-bae2-4603-b5ad-3bb497d9790d">
doris中确实没这个函数。

**Reply by HWTsang** [reply]
我们是从doris2.0开始用的, 目前升级到2.0.4了, 之前没有遇到这个bug, 使用正常, 前几天发现有一张表在metabase少了一个字段, 单独那个字段没有出现在metabase中, 还在排查, 用metabase的sql查可以查到

**Reply by ihadoop** [reply]
我那个错误我找到问题了，是最新版本metabase的mysql驱动改了，确实没法兼容doris了。你少的字段是什么类型的，我刚好在写doris的driver

**Reply by ihadoop** [reply]
如果方便，可以把少字段的表结构发我，我本地测试下

**Reply by HWTsang** [reply]
升级到2.0.5之后metabase报错, 可能要把metabase也升级一下才能解决

`
﻿[96a4b8ba-c581-4d20-afb8-7e54ed4be183] 2024-01-31T10:54:00+08:00 WARN metabase.sync.util 同步表"表669'ordds'"字段失败

java.sql.SQLTransientConnectionException: (conn=446) errCode = 2, detailMessage = (111.10.11.111)[CANCELLED][BUFFER_ALLOCATION_FAILED]concat output is too large to allocate

at org.mariadb.jdbc.internal.util.exceptions.ExceptionFactory.createException(ExceptionFactory.java:79)
......

Caused by: java.sql.SQLException: errCode = 2, detailMessage = (111.10.11.111)[CANCELLED][BUFFER_ALLOCATION_FAILED]concat output is too large to allocate

0#  doris::vectorized::FunctionStringConcat::execute_impl(doris::FunctionContext*, doris::vectorized::Block&, std::vector<unsigned long, std::allocator<unsigned long> > const&, unsigned long, unsigned long) at /home/zcp/repo_center/doris_release/doris/be/src/common/status.h:0

1#  doris::vectorized::DefaultExecutable::execute_impl(doris::FunctionContext*, doris::vectorized::Block&, std::vector<unsigned long, std::allocator<unsigned long> > const&, unsigned long, unsigned long) at /home/zcp/repo_center/doris_release/doris/be/src/vec/functions/function.h:517

2#  doris::vectorized::PreparedFunctionImpl::_execute_skipped_constant_deal(doris::FunctionContext*, doris::vectorized::Block&, std::vector<unsigned long, std::allocator<unsigned long> > const&, unsigned long, unsigned long, bool) at /home/zcp/repo_center/doris_release/doris/be/src/vec/functions/function.cpp:153

3#  doris::vectorized::PreparedFunctionImpl::default_implementation_for_nulls(doris::FunctionContext*, doris::vectorized::Block&, std::vector<unsigned long, std::allocator<unsigned long> > const&, unsigned long, unsigned long, bool, bool*) at /home/zcp/repo_center/doris_release/doris/be/src/vec/functions/function.cpp:268

4#  doris::vectorized::PreparedFunctionImpl::execute_without_low_cardinality_columns(doris::FunctionContext*, doris::vectorized::Block&, std::vector<unsigned long, std::allocator<unsigned long> > const&, unsigned long, unsigned long, bool) at /home/zcp/repo_center/doris_release/doris/be/src/vec/functions/function.cpp:0

5#  doris::vectorized::PreparedFunctionImpl::execute(doris::FunctionContext*, doris::vectorized::Block&, std::vector<unsigned long, std::allocator<unsigned long> > const&, unsigned long, unsigned long, bool) at /home/zcp/repo_center/doris_release/doris/be/src/vec/functions/function.cpp:274

6#  doris::vectorized::IFunctionBase::execute(doris::FunctionContext*, doris::vectorized::Block&, std::vector<unsigned long, std::allocator<unsigned long> > const&, unsigned long, unsigned long, bool) at /home/zcp/repo_center/doris_release/doris/be/src/vec/functions/function.h:183

7#  doris::vectorized::VectorizedFnCall::execute(doris::vectorized::VExprContext*, doris::vectorized::Block*, int*) at /home/zcp/repo_center/doris_release/doris/be/src/vec/exprs/vectorized_fn_call.cpp:159

8#  doris::vectorized::VectorizedFnCall::execute(doris::vectorized::VExprContext*, doris::vectorized::Block*, int*) at /home/zcp/repo_center/doris_release/doris/be/src/common/status.h:442

9#  doris::vectorized::VectorizedFnCall::execute(doris::vectorized::VExprContext*, doris::vectorized::Block*, int*) at /home/zcp/repo_center/doris_release/doris/be/src/common/status.h:442

10# doris::vectorized::VectorizedFnCall::execute(doris::vectorized::VExprContext*, doris::vectorized::Block*, int*) at /home/zcp/repo_center/doris_release/doris/be/src/common/status.h:442

11# doris::vectorized::VExprContext::execute(doris::vectorized::Block*, int*) at /home/zcp/repo_center/doris_release/doris/be/src/common/status.h:354

12# doris::vectorized::VExprContext::get_output_block_after_execute_exprs(std::vector<std::shared_ptr<doris::vectorized::VExprContext>, std::allocator<std::shared_ptr<doris::vectorized::VExprContext> > > const&, doris::vectorized::Block const&, doris::vectorized::Block*, bool) at /home/zcp/repo_center/doris_release/doris/be/src/common/status.h:442

13# doris::vectorized::VMysqlResultWriter<false>::append_block(doris::vectorized::Block&) at /home/zcp/repo_center/doris_release/doris/be/src/common/status.h:442

14# doris::vectorized::VResultSink::send(doris::RuntimeState*, doris::vectorized::Block*, bool) at /home/zcp/repo_center/doris_release/doris/be/src/common/status.h:442

15# doris::pipeline::DataSinkOperator<doris::pipeline::ResultSinkOperatorBuilder>::sink(doris::RuntimeState*, doris::vectorized::Block*, doris::pipeline::SourceState) at /home/zcp/repo_center/doris_release/doris/be/src/pipeline/exec/operator.h:0

16# doris::pipeline::PipelineTask::execute(bool*) at /home/zcp/repo_center/doris_release/doris/be/src/common/status.h:437

17# doris::pipeline::TaskScheduler::_do_work(unsigned long) at /home/zcp/repo_center/doris_release/doris/be/src/common/status.h:354

18# doris::ThreadPool::dispatch_thread() at /home/zcp/repo_center/doris_release/doris/be/src/util/threadpool.cpp:0

19# doris::Thread::supervise_thread(void*) at /var/local/ldb_toolchain/bin/../usr/include/pthread.h:562

20# start_thread

21# __GI___clone

at org.mariadb.jdbc.internal.protocol.AbstractQueryProtocol.readErrorPacket(AbstractQueryProtocol.java:1695)

at org.mariadb.jdbc.internal.protocol.AbstractQueryProtocol.readPacket(AbstractQueryProtocol.java:1557)

at org.mariadb.jdbc.internal.protocol.AbstractQueryProtocol.getResult(AbstractQueryProtocol.java:1520)

at org.mariadb.jdbc.internal.protocol.AbstractQueryProtocol.executeQuery(AbstractQueryProtocol.java:256)

... 91 more`

**Reply by ihadoop** [reply]
metabase你要是用最新版，压根就没法访问doris了

**Reply by HWTsang** [reply]
确实, 我升级metabase以后也是报你那个current_user的错, 同时老的metabase不能连最新的doris2.0.4了, 现在得回退doris到2.0.3同时用老的metabase才能操作

**Reply by ihadoop** [reply]
我写好了driver，本地测试可以最新的metabase链接doris

**Reply by HWTsang** [reply]
在哪测, 会放到官方上面吗

**Reply by ihadoop** [reply]
方便留个联系方式吗？我发你jar。我提issue，看doris怎么处理吧

[doris.metabase-driver.zip](https://github.com/apache/doris/files/14125756/doris.metabase-driver.zip)

下载后，把后缀zip换成jar

**Reply by HWTsang** [reply]
我们现在是用的doris2.0.3+最新的metabase, 虽然metabase会有一点报错,但是不影响使用

**Reply by ihadoop** [reply]
最新的metabase是连不上doris的，metabase逻辑变了

**Comment by Matthrew** [comment]
> 最新的metabase是连不上doris的，metabase逻辑变了
您好，请问下，我用你这个连接doris驱动，没有表的关联，联合join这个功能，这个是什么原因呢？

**Comment by HWTsang** [comment]
之前最新的metabase又可以连上doris2.04了
